Backoffice/EventSubscriber/BlockFormTypeSubscriber.php

```diff
@@ -35,6 +36,20 @@ public function __construct(array $fixedParameters)
     /**
      * @param FormEvent $event
      */
+    public function preSetData(FormEvent $event)
+    {
+        $form = $event->getForm();
+        $data = $event->getData();
+        $label = $data->getLabel();
+        $blockPosition = $form->getConfig()->getOption('blockPosition');
+        if ('' == $label && null !== $blockPosition) {
+            $data->setLabel($data->getComponent() . ' #'. ($blockPosition +1));
```

Comment:
Instead of coding here the label generation, maybe you could use a dedicated protected method ``generateLabel($data, $blockposition)`` to isolate the logic